gradle.properties

```diff
@@ -1,25 +1,22 @@
 org.gradle.jvmargs=-Xms1g -Xmx4g -Dfile.encoding=UTF-8 -XX:MaxMetaspaceSize=1g
 
-# When configured, Gradle will run in incubating parallel mode.
-# This option should only be used with decoupled projects. More details, visit
-# http://www.gradle.org/docs/current/userguide/multi_project_builds.html#sec:decoupled_projects
 org.gradle.parallel=true
-
-# Disabled as this has no benefits in studio builds and only marginal benefits in command line, but
-# can cause problems with Kotlin Gradle DSL. We're observing this for a week to see if/how it affects
-# metrics.
-org.gradle.configureondemand=false
-
-# Enable remote build cache
-# https://docs.gradle.org/current/userguide/build_cache.html
+org.gradle.configureondemand=true
 org.gradle.caching=true
 
+# New Kotlin IC flags
+kotlin.compiler.suppressExperimentalICOptimizationsWarning=true
+kotlin.compiler.keepIncrementalCompilationCachesInMemory=true
+kotlin.compiler.preciseCompilationResultsBackup=true
+
 # Necessary for the intellij plugin but also not helpful for Gradle plugins since Gradle forces its own
 kotlin.stdlib.default.dependency=false
 
 # TODO temporary workaround for Kotlin 1.8.20+ (https://jb.gg/intellij-platform-kotlin-oom)
```

Comment:
Also looks like this can be removed in this new version 

https://plugins.jetbrains.com/docs/intellij/using-kotlin.html#incremental-compilation

Remove additional kotlin.incremental.useClasspathSnapshot=false property in gradle.properties if present